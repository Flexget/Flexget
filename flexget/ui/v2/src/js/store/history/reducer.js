import { GET_HISTORY } from 'store/history/actions';

const initState = {
  totalCount: 0,
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
