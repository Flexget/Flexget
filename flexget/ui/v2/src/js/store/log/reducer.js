import {
  LOG_CONNECT,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'store/log/actions';

const initState = {
  messages: [],
  connected: false,
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_CONNECT:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
