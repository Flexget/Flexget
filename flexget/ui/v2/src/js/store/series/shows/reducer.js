import { GET_SHOWS } from 'store/series/shows/actions';

const initState = {
  totalCount: 0,
  items: [],
};

export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_SHOWS:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.data,
      };
    default:
      return state;
  }
}
