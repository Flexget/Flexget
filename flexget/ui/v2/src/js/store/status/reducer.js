import {
  LOADING_STATUS,
  ERROR_STATUS,
  CLOSE_STATUS,
  INFO_STATUS,
} from 'store/status/actions';
import { LOCATION_CHANGE } from 'connected-react-router';

const initState = {
  loading: {},
  info: null,
  error: null,
};

export default (state = initState, action) => {
  switch (action.type) {
    case LOADING_STATUS:
      return {
        ...state,
        loading: {
          ...state.loading,
          [action.meta.type]: true,
        },
      };
    case ERROR_STATUS: {
      const {
        [action.payload.type]: omit,
        ...loading
      } = state.loading;
      return {
        ...state,
        error: action.payload,
        loading,
      };
    }
    case INFO_STATUS:
      return {
        ...state,
        info: action.payload.message,
      };
    case CLOSE_STATUS:
    case LOCATION_CHANGE:
      return {
        ...initState,
        loading: state.loading,
      };
    default:
      if (state.loading[action.type]) {
        const {
          [action.type]: omit,
          ...loading
        } = state.loading;
        return {
          ...state,
          loading,
        };
      }
      return state;
  }
};
