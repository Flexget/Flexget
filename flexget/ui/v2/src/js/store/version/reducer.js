import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case GET_VERSION: {
      const {
        api_version: api,
        flexget_version: flexget,
        latest_version: latest,
      } = action.payload;

      return { api, flexget, latest };
    }
    default:
      return state;
  }
}
