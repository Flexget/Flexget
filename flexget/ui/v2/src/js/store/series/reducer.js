import { combineReducers } from 'redux';
import shows from 'store/series/shows/reducer';

export default combineReducers({
  shows,
});
