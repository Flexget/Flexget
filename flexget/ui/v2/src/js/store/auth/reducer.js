import { LOGIN, LOGOUT } from 'store/auth/actions';
import { ERROR_STATUS } from 'store/status/actions';
import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
    case GET_VERSION:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
