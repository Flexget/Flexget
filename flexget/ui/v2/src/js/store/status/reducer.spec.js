import {
  LOADING_STATUS,
  ERROR_STATUS,
  CLOSE_STATUS,
  INFO_STATUS,
} from 'store/status/actions';
import { LOCATION_CHANGE } from 'connected-react-router';
import reducer from 'store/status/reducer';

const TEST = 'TEST';

describe('store/status/reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual({
      loading: {},
      info: null,
      error: null,
    });
  });

  describe('LOADING_STATUS', () => {
    it('should set loading status when initially empty', () => {
      expect(reducer(undefined, {
        type: LOADING_STATUS,
        meta: {
          type: TEST,
        },
      })).toMatchSnapshot();
    });

    it('should set loading status when not empty', () => {
      expect(reducer({
        loading: {
          TEST: true,
        },
        error: null,
        info: null,
      }, {
        type: LOADING_STATUS,
        meta: {
          type: 'OTHER',
        },
      })).toMatchSnapshot();
    });
    it('should not change if already loading that action', () => {
      expect(reducer({
        loading: {
          TEST: true,
        },
        error: null,
        info: null,
      }, {
        type: LOADING_STATUS,
        meta: {
          type: TEST,
        },
      })).toMatchSnapshot();
    });
  });

  it('should handle ERROR_STATUS', () => {
    expect(reducer(undefined, {
      type: ERROR_STATUS,
      payload: {
        message: 'something',
      },
    })).toMatchSnapshot();
  });

  it('should handle INFO_STATUS', () => {
    expect(reducer(undefined, {
      type: INFO_STATUS,
      payload: {
        message: 'something',
      },
    })).toMatchSnapshot();
  });

  it('should clear state on CLOSE_STATUS', () => {
    expect(reducer(undefined, { type: CLOSE_STATUS })).toMatchSnapshot();
  });

  it('should clear state on LOCATION_CHANGE', () => {
    expect(reducer(undefined, { type: LOCATION_CHANGE })).toMatchSnapshot();
  });

  it('should handle loding done', () => {
    expect(reducer({
      loading: {
        TEST: true,
      },
      error: null,
      info: null,
    }, {
      type: TEST,
    })).toMatchSnapshot();
  });
});
