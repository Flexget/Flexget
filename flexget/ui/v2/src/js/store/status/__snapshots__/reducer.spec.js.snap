// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/status/reducer LOADING_STATUS should not change if already loading that action 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {
    "TEST": true,
  },
}
`;

exports[`store/status/reducer LOADING_STATUS should set loading status when initially empty 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {
    "TEST": true,
  },
}
`;

exports[`store/status/reducer LOADING_STATUS should set loading status when not empty 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {
    "OTHER": true,
    "TEST": true,
  },
}
`;

exports[`store/status/reducer should clear state on CLOSE_STATUS 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {},
}
`;

exports[`store/status/reducer should clear state on LOCATION_CHANGE 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {},
}
`;

exports[`store/status/reducer should handle ERROR_STATUS 1`] = `
Object {
  "error": Object {
    "message": "something",
  },
  "info": null,
  "loading": Object {},
}
`;

exports[`store/status/reducer should handle INFO_STATUS 1`] = `
Object {
  "error": null,
  "info": "something",
  "loading": Object {},
}
`;

exports[`store/status/reducer should handle loding done 1`] = `
Object {
  "error": null,
  "info": null,
  "loading": Object {},
}
`;
